File: app/src/main/java/me/zhanghai/android/files/provider/content/ContentFileAttributesImpl.java
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.provider.content;

import android.net.Uri;

import org.threeten.bp.Instant;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import java8.nio.file.attribute.FileTime;
import me.zhanghai.android.files.provider.common.ContentProviderFileAttributes;

class ContentFileAttributesImpl implements ContentProviderFileAttributes {

    @Nullable
    private final String mMimeType;
    private final long mSize;
    @NonNull
    private final Uri mUri;

    ContentFileAttributesImpl(@Nullable String mimeType, long size, @NonNull Uri uri) {
        mMimeType = mimeType;
        mSize = size;
        mUri = uri;
    }

    @NonNull
    @Override
    public FileTime lastModifiedTime() {
        return FileTime.from(Instant.EPOCH);
    }

    @NonNull
    @Override
    public FileTime lastAccessTime() {
        return lastModifiedTime();
    }

    @NonNull
    @Override
    public FileTime creationTime() {
        return lastModifiedTime();
    }

    @Nullable
    @Override
    public String mimeType() {
        return mMimeType;
    }

    @Override
    public long size() {
        return mSize;
    }

    @NonNull
    @Override
    public Uri fileKey() {
        return mUri;
    }
}

File: app/src/main/java/me/zhanghai/android/files/provider/content/ContentFileAttributesImpl.kt
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.files.provider.content

import android.net.Uri
import java8.nio.file.attribute.FileTime
import me.zhanghai.android.files.provider.common.ContentProviderFileAttributes
import org.threeten.bp.Instant

internal class ContentFileAttributesImpl(
    private val mimeType: String?,
    private val size: Long,
    private val uri: Uri
) : ContentProviderFileAttributes {
    override fun lastModifiedTime(): FileTime = FileTime.from(Instant.EPOCH)

    override fun lastAccessTime(): FileTime = lastModifiedTime()

    override fun creationTime(): FileTime = lastModifiedTime()

    override fun mimeType(): String? = mimeType

    override fun size(): Long = size

    override fun fileKey(): Uri = uri
}
